akka-actor-testkit-typed/src/test/java/jdocs/akka/actor/testkit/typed/javadsl/AsyncTestingExampleTest.java

```diff
@@ -39,9 +40,9 @@
   public static class Echo {
     public static class Ping {
       public final String message;
-      public final ActorRef<Pong> replyTo;
+      public final RecipientRef<Pong> replyTo;
```

Comment:
While it is good that this is covered in docs I don't expect that it will be how people really use it much, given that it basically means adapting the message protocol specifically to avoid `probe.ref` in the tests. 

So maybe it is misleading to show it off like this, better to keep this sample as is but instead only mention it in the text next to where this code example is shown off?

Reply:
I personally have taken to using `RecipientRef` where possible in my protocols and states (if for no other reason than to give me more freedom to switch to an `entityRef`, though principle of least power); that's a big part of what motivated me looking into this.

Reply:
I think that is a fine choice, but I'm not convinced that flexibility is important enough that we should push it on new users learning the APIs/reading the docs.